astroquery/esa/euclid/core.py

```diff
@@ -1175,9 +1186,19 @@ def get_spectrum(self, *, source_id, schema='sedm', output_file=None, verbose=Fa
         if source_id is None or schema is None:
             raise ValueError(self.__ERROR_MSG_REQUESTED_GENERIC)
 
-        params_dict = {'TAPCLIENT': 'ASTROQUERY', 'RETRIEVAL_TYPE': 'SPECTRA'}
+        rt = str(retrieval_type).upper()
+        if rt != 'ALL' and rt not in self.__VALID_DATALINK_RETRIEVAL_TYPES:
+            raise ValueError(f"Invalid mandatory argument 'retrieval_type'. Found {retrieval_type}, "
```

Comment:
docstring says this is optional, so you may want to rephrase it.
```suggestion
            raise ValueError(f"Invalid argument value for 'retrieval_type'. Found {retrieval_type}, "
```